wizeline/project-lab
# Feat - #247 - Practiced Skills to Project Members

#### What does this PR do?

Added relation between projectMember and skills. Now, when joining a project, you will be asked to input which skills you are going to practice. Right now, you can edit these skills (for all project members) from the Edit Project view.

#### Where should the reviewer start?

- db/schema.prisma
- app/core/components/SkillsSelect/index.tsx
- app/core/components/ProjectMembersField/index.tsx
- app/projects/mutations/createProjectMember.ts
- app/projects/mutations/updateProject.ts
- app/projects/components/joinProjectModal/index.tsx
- app/projects/validations.ts

#### How should this be manually tested?

1. Run new migration.
2. Run project locally.
3. Log in.
4. Try to join a new project.
5. In the join modal, you should be prompted to input skills to practice as well.
6. You should join successfully.
7. Go to Edit the project
8. You should see yourself in the project members, now showing a field for your skills.
9. Try to edit your skills, then save.
10. If you enter the Edit Project view after saving changes, you should see your new skills reflected.

#### Any background context you want to provide?

N/A

#### What are the relevant tickets?

#247 

#### Screenshots

<img width="837" alt="image" src="https://user-images.githubusercontent.com/97301313/160026897-6d06bd5f-fb6b-465f-b564-ce209dbe1529.png">
_Join project modal_
<img width="1272" alt="image" src="https://user-images.githubusercontent.com/97301313/160026985-611635c2-976f-45d8-9b39-c3660d8b8d87.png">
_Edit Project view: project members_

#### Questions

N/A

#### Notes

- Minor changes made to the projectMembersField layout. 
- Added props to the SkillsSelect to be more reusable / customisable (defaultValue = [] in order to avoid an error, fullWidth, style, size, customOnChange (in case you want to provide an extra/custom functionality to be executed on change)
- Updated seeds, just to have an initial member with practicedSkills in Project Lab
- For future improvements, as a suggestion, a user's practiced skills should be displayed in a User/My Account/Profile page, to have in a same view ALL the skills being practiced in ALL the different projects that a User may be a member of.

#### Checklist
